File: content-interface/src/main/java/com/service/CartService.java
package com.service;

import com.shopxx.pojo.XxCart;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CartService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<XxCart> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(XxCart cart);
	
	
	/**
	 * 修改
	 */
	public void update(XxCart cart);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public XxCart findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(XxCart cart, int pageNum, int pageSize);
	
}

File: manager-web/src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../cart/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../cart/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../cart/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../cart/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../cart/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../cart/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../cart/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
